components/reusable-components/reusable-table/FetchData.tsx

```diff
@@ -0,0 +1,73 @@
+export interface User {
+  id: string;
+  first_name: string;
+  last_name: string;
+  role: string;
+}
+
+const url = 'http://localhost:3001/users';
+
+const headers = {
+  'Content-Type': 'application/json',
+  Accept: 'application/json',
+};
```

Comment:
we should fetch the blog posts here, not users. Please do not hardcore urls.
For security reasons we should keep the URLs in the .env file.
In the code we will use them as :
const url = process.env.NEXT_PUBLIC_API_BASE_URL + '/content/blog-posts'!;
or use dummy data so we can see and test the table

Reply:
Also use more descriptive names for all functions in this file